### PR TITLE
Update to Zivid SDK v2.1.0 and modify downsample sample

### PR DESCRIPTION
**Pull request by robertkt1**
Almost zero code to write. Deleted a lot of code and changed a couple of lines to match built-in function

### CODE REVIEW

File: source/Applications/Advanced/Downsample/Downsample.cpp
Comment (SatjaSivcev) [inline source/Applications/Advanced/Downsample/Downsample.cpp:41]: Add the following line before
`std::cout << "Downsampling point cloud" << std::endl;`

File: source/Applications/Advanced/Downsample/Downsample.cpp
Comment (SatjaSivcev) [inline source/Applications/Advanced/Downsample/Downsample.cpp:42]: One liner? No need for downsamplingFactor variable I think

**Review by SatjaSivcev — CHANGES_REQUESTED** [review]
Minor changes required.
Have you tested that this works ok?
P.S. You will need one PR before this one where you modify:
- CMakeLists - set(ZIVID_VERSION 2.0.0)
- CI files

**Comment by SatjaSivcev** [comment]
In Readme.md you have 2.0.0 hardcoded. Fix to 2.1.0

Tip: Open the whole project in Visual Studio Code and search for 2.0.

**Review by SatjaSivcev — COMMENTED** [review]
One last comment to fix. We can wait until the release candidate now.

**Comment by SatjaSivcev** [comment]
> 
> 
> One last comment to fix. We can wait until the release candidate now.

This is done. Waiting for the release now.

**Comment by SatjaSivcev** [comment]
It seems this sample is using 'fileName ' and other ones that load zdf use 'dataFile'. Can you check and confirm and if true modifiy this one?

**Comment by SatjaSivcev** [comment]
You can rebase on master and squash commits. Perhaps it is best to have a single commit message called stg like 'Update samples for SDK 2.1.0' (check commit history to see what we did), and in that commit you can specify what has been done (This commit modified Downsample sample, etc...

File: source/Applications/Advanced/Downsample/Downsample.cpp
Comment (runenordmo) [inline source/Applications/Advanced/Downsample/Downsample.cpp:42]: We also support in-place downsampling; `downsample()`. I think that is more relevant for the sample, since that is more performant and should be the default.

An good alternative is to show both in-place downsampling (`downsample()`) and downsampling into a new `Pointcloud` instance (`downsampled()`).

**Comment by runenordmo** [comment]
LGTM, will likely approve when my comment on doing the downsampling in-place is addressed.

File: source/Applications/Advanced/Downsample/Downsample.cpp
Comment (eskaur) [reply source/Applications/Advanced/Downsample/Downsample.cpp:42]: Do you suggest moving the first visualizer call to before "downsample" then?

Comment (robertkt1) [reply source/Applications/Advanced/Downsample/Downsample.cpp]: That's how I solved it now. By moving the first viz

**Review by runenordmo — APPROVED** [review]
LGTM :+1: Can you just rebase and squash the two commits into one?
You can also test that the sample still works (win/ubuntu)

**Comment by runenordmo** [comment]
@robertkt1, there is a spelling error in the commit message: 
`Update downsamnple sample for SDK 2.1`

**Review by runenordmo — CHANGES_REQUESTED** [review]
You need to correct the commit message spelling error:
`Update downsamnple sample for SDK 2.1`

File: source/Applications/Advanced/Downsample/Downsample.cpp
Comment (SatjaSivcev) [reply source/Applications/Advanced/Downsample/Downsample.cpp]: > 
> 
> We also support in-place downsampling; `downsample()`. I think we should rather use that in this sample, since it is more performant (no copy) and should definitely be the default. @robertkt1
> 
> An alternative is to show both in-place downsampling (`downsample()`) and downsampling into a new `Pointcloud` instance (`downsampled()`).

@robertkt1 Rune's comment on two options should be covered in the point cloud tutorial: https://github.com/zivid/zivid-cpp-samples/blob/new-downsample-sample/source/Applications/PointCloudTutorial.md

**Comment by runenordmo** [comment]
You should format your code in `Downsample.cpp`, @robertkt1. Run `clang-format -i -style=file <path_to>/Downsample.cpp`

File: source/Applications/Advanced/Downsample/Downsample.cpp
Comment (runenordmo) [inline source/Applications/Advanced/Downsample/Downsample.cpp:39]: Nitpick:
The output of this sample is now:
```
1. Size of point cloud before downsampling: 2861568 data points
2. Visualizing point cloud before downsampling
3. Setting up visualization
4. Visualizing point cloud
5. Running visualizer. Blocking until window closes
6. Downsampling point cloud
7. Size of point cloud after downsampling: 178848 data points
8. Visualizing point cloud after downsampling
9. Setting up visualization
10. Visualizing point cloud
11. Running visualizer. Blocking until window closes
```
**Line 2-5 and 8-11 is basically the same information given four lines in a row.
I think you can remove two of the printouts from inside `visualizePointCloud()`,** given that is does not give any inconsistency with the other samples.

**Review by runenordmo — COMMENTED** [review]
This looks close to merge to me now :+1:  I added only one nitpick comment. When you answer that, I'll approve.

Btw, have you tested running the samples for 2.1 (should be no reason they should not)? It looks like the CI pipeline only test that the samples build.

**Review by runenordmo — APPROVED** [review]
Nice. Now you can squash your fixup commit into the original commit, then rebase on master and merge when CI's green :+1:

**Review by SatjaSivcev — APPROVED** [review]

**Pull request merged**
